src/app/register/register.component.html

```diff
@@ -0,0 +1,82 @@
+<h3 i18n>Become a member</h3>
+<form [formGroup]="registerForm" (ngSubmit)="registerUser(registerForm.value)">
+  <mat-grid-list cols="3">
+    <mat-form-field>
+      <input matInput i18n-placeholder placeholder="First Name" formControlName="firstName" required>
+      <mat-error><planet-form-error-messages [control]="registerForm.controls.firstName"></planet-form-error-messages></mat-error>
+    </mat-form-field>
+    <mat-form-field>
+      <input matInput i18n-placeholder placeholder="Middle Name" formControlName="middleName" required>
+      <mat-error><planet-form-error-messages [control]="registerForm.controls.middleName"></planet-form-error-messages></mat-error>
+    </mat-form-field>
+    <mat-form-field>
+      <input matInput i18n-placeholder placeholder="Last Name" formControlName="lastName" required>
+      <mat-error><planet-form-error-messages [control]="registerForm.controls.LastName"></planet-form-error-messages></mat-error>
+    </mat-form-field>
+  </mat-grid-list>
+  <mat-grid-list cols="3">
+    <mat-form-field>
+      <input matInput i18n-placeholder placeholder="Login" formControlName="login" required>
```

Comment:
please do check username, and make user name unique, same user name can't be register twice.

Reply:
@amanuely You can use ValidatorService(validator.service.ts) to check unique field.

Reply:
@nirojdyola  I tried to use ValidatorService, but it is not working for me. I followed the example in course_add_component.ts. Could you please have a look at the file and see where I am making mistakes? Thank you!!

Reply:
@amanuely It can be used for _users . So what you can do is look createUser function of login component . This function throws error when same login name is added. You can check with browser inspect element, so similarly you can do it here. You can catch this error for login name unique.

Reply:
@nirojdyola  Thank you. That is very useful.